### PR TITLE
[WIP] SRVKE-1419: Document Eventing mTLS with Service Mesh

### PR DESCRIPTION
**Pull request by pierDipi**
Version(s):

Issue:

Link to docs preview:

- https://58391--docspreview.netlify.app/openshift-enterprise/latest/serverless/integrations/serverless-ossm-setup.html#serverless-ossm-setup_serverless-ossm-setup

QE review:
- [ ] QE has approved this change.

Additional information:

### CODE REVIEW

File: modules/serverless-ossm-setup.adoc [hunk 1]
```diff
@@ -183,6 +184,147 @@ spec:
...
+kind: KnativeEventing
+metadata:
+  name: knative-eventing
+  namespace: knative-eventing
+spec:
+  config:
+    features:
+      istio: enabled
+  workloads: <1>
+  - name: pingsource-mt-adapter
```
Comment (matzew) [inline modules/serverless-ossm-setup.adoc:201]: should we say that we use the ping source as an example here?

File: modules/serverless-ossm-setup.adoc [hunk 2]
```diff
@@ -183,6 +184,147 @@ spec:
...
+    annotations:
+      "sidecar.istio.io/inject": "true"
+      "sidecar.istio.io/rewriteAppHTTPProbers": "true"
+----
+<1> Enables Eventing istio controller to create a `DestinationRule` for each InMemoryChannel or KafkaChannel service.
+<2> Enables sidecar injection for Knative Eventing pods.
```
Comment (matzew) [inline modules/serverless-ossm-setup.adoc:207]: is the `<2>` missing in the above?

File: modules/serverless-ossm-setup.adoc [hunk 3]
```diff
@@ -183,6 +184,147 @@ spec:
...
+    annotations:
+      "sidecar.istio.io/inject": "true"
+      "sidecar.istio.io/rewriteAppHTTPProbers": "true"
+  - name: kafka-sink-receiver
+    annotations:
+      "sidecar.istio.io/inject": "true"
+      "sidecar.istio.io/rewriteAppHTTPProbers": "true"
+----
+<1> The Apache Kafka cluster URL, for example: `my-cluster-kafka-bootstrap.kafka:9092`.
+<2> Enables sidecar injection for Knative Kafka pods.
```
Comment (matzew) [inline modules/serverless-ossm-setup.adoc:265]: Knative pods for Apache KAfka?

File: modules/serverless-ossm-setup.adoc [hunk 4]
```diff
@@ -183,6 +184,147 @@ spec:
 $ oc apply -f <filename>
 ----
 
+. Install Knative Eventing by creating the following `KnativeEventing` custom resource definition (CRD), which also enables the Istio integration:
```
Comment (matzew) [inline modules/serverless-ossm-setup.adoc:187]: maybe we can have this (and serving, and knative kafka), as sub items on the Navigation bar in the documentation?

File: modules/serverless-ossm-setup.adoc (hunk 2 above)
Comment (pierDipi) [reply modules/serverless-ossm-setup.adoc:207]: Yes, good catch!

File: modules/serverless-containersource-ossm.adoc [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+spec:
+  template:
+    metadata: <2>
+      annotations:
+        sidecar.istio.io/inject": "true"
+        sidecar.istio.io/rewriteAppHTTPProbers: "true"
+    spec:
+      containers:
+        # This corresponds to a heartbeats image URI that you have built and published
+        - image: gcr.io/knative-releases/knative.dev/eventing/cmd/heartbeats
```
Comment (matzew) [inline modules/serverless-containersource-ossm.adoc:64]: perhaps `quay.io/openshift-knative/heartbeats:latest`  ?

File: modules/serverless-ossm-setup.adoc (hunk 1 above)
Comment (matzew) [reply modules/serverless-ossm-setup.adoc:201]: Perhaps add that, somehow?

File: modules/serverless-ossm-setup.adoc (hunk 3 above)
Comment (pierDipi) [reply modules/serverless-ossm-setup.adoc:265]: Good point

File: modules/serverless-containersource-ossm.adoc (hunk 5 above)
Comment (pierDipi) [reply modules/serverless-containersource-ossm.adoc:64]: I think I copied it from somewhere else :thinking:, so I'll change both to use quay

File: modules/serverless-ossm-setup.adoc [hunk 6]
```diff
@@ -183,6 +184,147 @@ spec:
...
+----
+apiVersion: operator.knative.dev/v1beta1
+kind: KnativeEventing
+metadata:
+  name: knative-eventing
+  namespace: knative-eventing
+spec:
+  config:
+    features:
+      istio: enabled
```
Comment (maschmid) [inline modules/serverless-ossm-setup.adoc:199]: is `config.features.istio` in KnativeEventing a thing now? (my impression from the design doc was that this idea was rejected)

Comment (pierDipi) [reply modules/serverless-ossm-setup.adoc:199]: Yes, we need this for https://issues.redhat.com/browse/SRVKE-1446 to enable/disable the controller, at the design doc time SRVKE-1446 wasn't explored, feel free to propose alternatives

**Pull request closed without merging**